unit_scaling/functional.py

```diff
@@ -247,52 +365,61 @@ def residual_split(input: Tensor, tau: float = 0.5) -> Tuple[Tensor, Tensor]:
     The tau factor allows unit scaling to behave as though the branches have different
     scales.
 
-    For MLP layers tau=0.5 is recommended, and for self-attention layers tau=0.01.
-
-    These values reflect the relative scales of the skip-vs-residual branches in a
-    standard transformer. Empirically, the self-attention tau is fairly insensitive
-    (i.e. tau=0.1 or tau=0.001 work well), but the default tau=0.5 causes significant
-    degradation.
-
     Args:
         input (Tensor): the tensor to which the residual layer is to be applied.
-        tau (float, optional): the weighting of the residual branch relative to the skip
-            connection. Defaults to 0.5.
+        tau (float, optional): the ratio of scale of contributions of the residual
+            branch to the skip connection. Larger values favor skip over residual.
+            Defaults to 1 (equal contribution).
 
     Returns:
         Tuple[Tensor, Tensor]: resulting tensors in the order: `residual, skip`.
     """
-    residual = scale_bwd(input, tau**0.5)
-    skip = scale_bwd(input, (1 - tau) ** 0.5)
+    denom = (1 + tau**2) ** 0.5
```

Comment:
Do you need to explain the `denom` in the docstring? Or is this just a new way to scale so that the default can be 1.0?

Reply:
Reading through the paper I think I understand this better now. It might be useful to explicitly reference the section in the paper justifying this parameterization (Appendix G I think?)

Reply:
:+1:, might leave it until the numbering is definitively frozen (very soon!), but yes.